specification/sbi_atee.adoc

```diff
@@ -13,6 +43,858 @@ running a VMM).
+    less than or equal to the value returned in tsm_info::tvm_max_vcpus
+    */ 
+    unsigned long tvm_num_vcpus;
+    /* The base physical address of the confidential memory region to be used
+    to hold the TVM’s vCPU state. Must be page-aligned and must be at least
+    TsmInfo::tvm_bytes_per_vcpu * tvm_num_vcpus bytes in length, rounded up
+    to the nearest multiple of 4KiB */
+    unsigned long tvm_vcpu_addr;
+};
+----
+
+*Returns* the `tvm_guest_id` in sbiret.value on success. The `tvm_guest_id` can be used
+to uniquely reference the TVM in invocations of other sbi_tee functions.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_create_tvm_errors]
+.TEE Create TVM Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The TVM with `tvm_guest_id` was successfully created.
+| SBI_ERR_INVALID_PARAM | The `tvm_create_params` or `tvm_create_params_len` parameters were invalid.
+|===
+
+=== Function: Destroy TVM (FID #1)
+[source, C]
+-----
+struct sbiret sbi_tee_destroy_tvm(unsigned long tvm_guest_id);
+-----
+Destroys a confidential TVM previously created using `sbi_tee_create_tvm()`. The tvm_guest_id
+is the unique identifier that was previously returned by `sbi_tee_create_tvm()`. 
+
+Note that destroying the TVM does not automatically release memory donated to by the caller,
+and reclaimation requires an explicit call to `sbi_tee_reclaim_pages`.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_destroy_tvm_errors]
+.TEE Destroy TVM Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The TVM with `tvm_guest_id` was successfully destroyed.
+| SBI_ERR_INVALID_PARAM | The `tvm_guest_id` was invalid.
+|===
+
+=== Function: TEE TVM Add Page Table Pages (FID #2)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_add_page_table_pages(unsigned long tvm_guest_id,
+                                               unsigned long base_page_address,
+                                               unsigned long num_pages);
+-----
+Adds num_pages 4KiB pages of confidential memory starting at page_addr to the
+TVM's page-table page-pool. The tvm_guest_id is the unique identifier that was
+returned by a previous call to `sbi_tee_create_tvm()`. 
+
+Callers can invoke the API multiple times to add pages incrementally, but not after
+`sbi_tee_tvm_finalize()` is called.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tvm_add_page_table_pages_errors]
+.TEE TVM Add Page Table Pages Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The TVM with `tvm_guest_id` was successfully destroyed.
+| SBI_ERR_INVALID_PARAM | The `tvm_guest_id` was invalid.
+|===
```

Comment:
SBI_ERR_INVALID_ADDRESS here and all other following functions that uses an address in one of the parameter.

Reply:
I don't see it being resolved ?

Reply:
should we report the error case where some memory was already converted?

Reply:
Sure -- should this be just an "invalid parameter" or do we want to define a more specific error?

Reply:
Not sure if this was answered as there is a github issue for TEE specific error code. A more specific error would be helpful here as well.